File: Backend.spring/SCPEx/src/main/java/com/demo/playground/scpex/Models/ContractNegotiation.java
package com.demo.playground.scpex.Models;

import com.demo.playground.scpex.Models.Enums.GeneralStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@AllArgsConstructor
@Getter
@Setter
public class ContractNegotiation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    Long id;

    String title;
    String description;

    Date dateCreated = new Date();

    @Enumerated(EnumType.ORDINAL)
    private GeneralStatus status = GeneralStatus.PENDING;

    @ManyToOne
    @JoinColumn(name = "contract_sender", nullable = false)
    Trader sender;

    public ContractNegotiation() { }
}

File: Frontend.react/scpex-front/src/models/ContractNegotiation.ts
import User from "./UserType/User.ts"

export default class ContractNegotiation {
    id: number | null;
    title: string;
    description: string;
    dateCreated: string;
    sender: User | null;

    constructor(id: number | null = null, title: string, description: string, sender: User | null = null) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.dateCreated = new Date().toISOString();
        this.sender = sender;
    }
}
